CodingStandards/ProgramEntryPoint.cs

```diff
@@ -5,33 +5,48 @@
 using System.Text;
 using System.Threading.Tasks;
 
+/// <summary>
+/// Writes a celebratory new year message.
+/// </summary>
 class Program
 {
-        //This is the entry point for the program.
-        static int Main(string[] args)
-{
-    return writeMessage(dtNow, Console.Out);
-}
-
-        static DateTime dtNow = DateTime.Now;
-
-public static int writeMessage(DateTime CurrentDate, TextWriter writer)
-{
-    var intRetVal = 0; const Int32 MESSAGE_COUNT = 5; for (int MyIndex = 0; MyIndex < MESSAGE_COUNT; MyIndex++)
-    /*********************/
-    /* greater than 2016 */
-    /*********************/
-    if (CurrentDate.Year > 2016) writer.WriteLine("Happy New Year!");
-    else
-                writer.WriteLine("Hello From " + CurrentDate.Year);
-
-
-    //--------------------return the result value----------------------------
-
-
-
-
-    return intRetVal;
+    private static DateTime Now = DateTime.Now;
+
+    // This is the entry point for the program.
+    /// <summary>
+    /// Prints five new year messages.
+    /// </summary>
+    /// <param name="args">Arguments for the application.</param>
+    /// <returns>The integer 0.</returns>
+    static int Main(string[] args)
+    {
+        return WriteMessage(Now, Console.Out);
     }
-}
 
+    /// <summary>
+    /// Writes a new year message when the supplied date is past 2016.
+    /// </summary>
+    /// <param name="currentDate">The current date.</param>
+    /// <param name="writer">The textwriter that will be used to write the message.</param>
+    /// <returns>The integer 0.</returns>
+    public static int WriteMessage(DateTime currentDate, TextWriter writer)
+    {
+        int returnValue = 0;
+        const int messageCount = 5;
+        for (int myIndex = 0; myIndex < messageCount; myIndex++)
```

Comment:
Just a stylistic comment: Conventionally, consts are placed at the top of a method and separated from other lines by a single space. Also, for readability, you'd typically see a blank line before the for statement.